test/framework/kubernetes.go

```diff
@@ -285,3 +289,73 @@ func checkPodContainerHasEnvVariableWithValue(pod *corev1.Pod, containerName, en
 	}
 	return false
 }
+
+// GetIngressURI returns the ingress URI
+func GetIngressURI(namespace, serviceName string) (string, error) {
+	ingress, err := WaitForIngress(namespace, serviceName)
+	if err != nil || ingress == nil || len(ingress.Spec.Rules) == 0 {
+		return "", fmt.Errorf("Ingress %s does not exist in namespace %s: %v", serviceName, namespace, err)
+	}
+
+	return fmt.Sprintf("http://%s:80", ingress.Spec.Rules[0].Host), nil
+}
+
+// WaitForIngress waits for ingress by a service
+func WaitForIngress(namespace, serviceName string) (*v1beta1.Ingress, error) {
+	ingress := &v1beta1.Ingress{}
+	if exists, err :=
+		kubernetes.ResourceC(kubeClient).FetchWithKey(types.NamespacedName{Name: serviceName, Namespace: namespace}, ingress); err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, nil
+	} else {
+		return ingress, nil
+	}
+}
+
+// ExposeServiceOnKubernetes adds ingress CR to expose a service
+func ExposeServiceOnKubernetes(namespace, serviceName string) error {
+	host := serviceName + namespace
```

Comment:
I will be adding a new parameter to add a hostname suffix here. This new parameter is only for Kubernetes, as Openshfit is adding the hostname suffix automatically.